apps/api_web/test/api_web/event_stream_test.exs

```diff
@@ -93,14 +93,24 @@ defmodule ApiWeb.EventStreamTest do
       assert_receive_data()
 
       send(self(), {:error, ["filter[]", " is required"]})
-      assert {:error, {:ok, conn}} = receive_result(state)
+      assert {:close, conn} = receive_result(state)
       assert chunks(conn) =~ "filter[] is required"
       on_exit(fn -> assert_stopped(state.pid) end)
     end
+
+    test "closes the connection when notified the diff server is exiting", %{conn: conn} do
+      state = initialize(conn, @module)
+      assert_receive_data()
+
+      send(self(), :exiting)
```

Comment:
Rather than faking this, can you stop the DiffServer and see that the result is correct?

Reply:
Good idea, I'll give it a try!